mongo/integration/crud_helpers_test.go

```diff
@@ -1425,8 +1425,22 @@ func executeAdminCommand(mt *mtest.T, op *operation) {
 		return
 	}
 
+	rco := options.RunCmd()
+	rpVal, err := op.Arguments.LookupErr("readPreference", "mode")
+	if err == nil {
+		if rp, ok := rpVal.StringValueOK(); ok {
+			switch rp {
+			case "Primary":
+				rco.SetReadPreference(mtest.PrimaryRp)
+			case "Secondary":
+				fmt.Println("setting secondary!")
+				rco.SetReadPreference(mtest.SecondaryRp)
+			}
+		}
+	}
```

Comment:
The `rediscover-quickly-after-step-down` SDAM integration test now uses `runAdminCommand` to run `replSetFreeze` with `readPreference: { mode: secondary }`. Added support in the unified test runner for a `readPreference` argument on `runAdminCommand`.